src/processor/StructureDefinitionProcessor.ts

```diff
@@ -58,11 +60,16 @@ export abstract class AbstractSDProcessor {
           ValueSetRuleExtractor.process(element)
         );
       }
+      const { obeysRule, invariants } = ObeysRuleExtractor.process(element);
+      newRules.push(obeysRule);
+      newInvariants.push(...invariants);
```

Comment:
I'm not sure how I feel about this.  I understand that in formal FHIR definitions, invariants are inlined in each resource -- but in FSH, invariants have global scope.  They can be re-used between resources.

For this reason, I might consider having two extractors -- an `ObeysRuleExtractor` and a `InvariantExtractor`.  This way the `ObeysRuleExtractor` follows the same pattern as the others (returning just an array of rules) and would fit more cleanly in the `if/else` above.  It would also allow the `InvariantExtractor` (or code around the `InvariantExtractor`) to check for duplicate invariants and only push to the invariants if it's unique by key.

In fact, we might want some error handling to detect if somehow (in some way) two invariants w/ the same key had differing definitions.  In that case, I'd issue an error and indicate we're only keeping the first one.

Whether or not we split this into two extractors is probably a matter of opinion.  I'm glad to discuss it.  But either way I think we need to properly deduplicate invariants here.